net-dns/dnscrypt-proxy: drop old, bump to 2.0.0, clean up metadata.

gyakovlev opened the pull request:
bump to 2.0.0, a completely new rewrite in go  

Remove old inactive maintainer.  
That line was meant to be removed some time ago but sneaked in during one of previous rebases.  
Also clean up descriptions.

Please note there is already a similar PR #7133, @comio agreed to use mine ebuild and work on it here in this PR if required.

@comio, need some systemd testing as mine is limited, openrc works fine.

Closes: https://bugs.gentoo.org/647110
Closes: https://bugs.gentoo.org/643642

mgorny commented on net-dns/dnscrypt-proxy/metadata.xml:
```diff
@@ -1,19 +1,13 @@
 <!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
 <pkgmetadata>
-	<maintainer type="person">
-		<email>amesists@gmail.com</email>
```
Since this is his only package, please also add:

    Closes: https://bugs.gentoo.org/643642

mgorny replied:
Oh, and given the `metadata.xml` you're using for your other packages, you may also want to remove proxy-maint if you wish to only directly work with Lars.

mgorny commented on net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild:
```diff
@@ -0,0 +1,89 @@
+src_compile() {
+	# Create directory structure suitable for building
+	mkdir -p "src/${EGO_PN}" || die
+	rm -r "src/${EGO_PN}" || die
```
This is terribly confusing to the reader (why do you remove something you just created?!). Please do instead:

    mkdir -p "src/${EGO_PN%/*}" || die

mgorny commented on net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild:
```diff
@@ -0,0 +1,89 @@
+}
+
+src_install() {
+	default
```
What does the `default` do here? Any reason you're not calling `golang-build_src_install`?

mgorny commented on net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild:
```diff
@@ -0,0 +1,89 @@
+
+	dobin dnscrypt-proxy
+
+	insinto /etc/${PN}
```
Minor nit: please don't use `${PN}` in install paths, this reduces readability and gives false sense of security that it will continue working if you rename package in Gentoo.

mgorny commented on net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild:
```diff
@@ -0,0 +1,89 @@
+	doins "src/${EGO_PN}"/example-{blacklist.txt,cloaking-rules.txt,forwarding-rules.txt}
+
+	insinto "/usr/share/${PN}"
+	doins -r "utils/generate-domains-blacklists/"
```
The behavior of `foo/` is undefined in POSIX, please use `foo/.` (with a full stop) instead.

mgorny commented on net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild:
```diff
@@ -0,0 +1,89 @@
+		ewarn
+	fi
+
+	if [[ ${REPLACING_VERSIONS} ]] && [[ ${REPLACING_VERSIONS} == 1.* ]]; then
```
This wrongly presumes that `REPLACING_VERSIONS` can have only one value. It is actually a space-delimited list, so please process it as a list (e.g. using `for`).

mgorny commented on net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild:
```diff
@@ -0,0 +1,89 @@
+	fcaps_pkg_postinst
+
+	if ! use filecaps; then
+		ewarn
```
Please remove the preceding empty lines, so you don't end up having two empty lines between each section.

mgorny commented on net-dns/dnscrypt-proxy/files/dnscrypt-proxy.initd-r2:
```diff
@@ -0,0 +1,19 @@
+start_pre() {
+	checkpath -q -d -m 0775 -o "${command_user}" /var/cache/"${SVCNAME}"
+	checkpath -q -d -m 0775 -o "${command_user}" /var/log/"${SVCNAME}"
+}
```
Please fix your editor to create trailing newline.

gyakovlev commented:
I've addressed all issues except one, that may need clarification.
I don't call ````golang-build_src_install```` because it'll install all vendored libraries and sources into go-gentoo systemwide GOPATH, and we don't need/want it.
Upstream packaging is a bit weird, everything required for building is bundled in the repo.

mgorny commented:
That's ok with me. Without `default` there should be no confusion that you've accidentally called the wrong default. Are you sure you don't want to install any docs via `einstalldocs` or `dodoc`?

gyakovlev commented:
good catch, ````default```` was doing that. pushed again with ````einstalldocs```` call.
One question about keywords:
I had to drop _~arm_   since I haven't tested it and it's a completely new package technically.
````
  KEYWORDS.dropped              1
   net-dns/dnscrypt-proxy/dnscrypt-proxy-2.0.0.ebuild: arm
````
I do have a RPi lying nearby but I never used gentoo on it, but maybe I can test and re-add it later, as it should work.

Reading [devmanual keywording entry](https://devmanual.gentoo.org/keywording/), it's a bit unclear if I should file a bug in this situation.

mgorny commented:
Yes, you should file a keywording bug after the package is committed so that ARM arch team can test the new version.

I have no further comments on this. I think it's ready to be committed. However, since on your other packages proxy-maint team is not listed and Lars has been pretty hostile towards me, I would like you to confirm whether you want proxy-maint to push for you, or if you are going to push your commits through Lars. If the former, please let me know and I'll merge it. If the latter, please remove proxy-maint from `metadata.xml` to avoid confusion.

comio commented:
Ok for me. I suggest to use systemd dynamic user if you have a recent systemd, but a separate user is also fine. I added the following:

```
[Service]
DynamicUser=yes
RuntimeDirectory=dnscrypt-proxy

```

ciao 

luigi

gyakovlev commented:
@mgorny Let's get it merged with proxy-maint in metadata. Lars is quite busy now and this package is popular and needs merging sooner, since old version is practically dead and rotting without maintenance.
@comio I'm not sure about dynamic users at least for now.
Since we create a user/group for openrc anyway why not just re-use it.
There are only 2 packages in portage to use it according to ````grep DynamicUser````  
I'll ask dev@g.o about this and add it later if no problems arise.

lucascouts commented:
I tried here, but the service doesn't work on openrc.

but this new version is amazing

mgorny commented:
@gyakovlev, ok. Please let me know if I should merge it or wait for you to resolve the OpenRC issue.

gyakovlev commented:
@mgorny WORKSFORME in _default_ configuration on several systems, including my apu2c4 in a paranoid   configuration, so let's merge  
@lucascouts file a bug after this PR is merged and provide logs and config files, I'll gladly review and help solve the problem if it does not work. If it's ebuild or init file problem I'll get it fixed ASAP.

dev-dvl commented:
Hello, 
Check
https://github.com/jedisct1/dnscrypt-proxy/issues/124

gyakovlev commented:
@dev-dvl Provided initscripts (openrc and systemd) pass -config /etc/dnscrypt-proxy/dnscrypt-proxy.toml to the executable, so the file is read by the binary.  

If you launch it manually it's up to you to point it to the right config file. Frank explained this in the mentioned issue, I see no problem here, or maybe you can explain what exactly bothers you.

On the other hand the issue made me realize that location of blacklists are relative to config file, I should not patch the paths to those in the config files. I'll do a revbump later, since it's not a breaking issue, just some polishing.

gyakovlev commented:
@dev-dvl Also check https://github.com/jedisct1/dnscrypt-proxy/wiki/installation  
it has gentoo specific section I've added some time ago.
